sources/packages/api/src/route-controllers/education-program-offering/education-program-offering.controller.ts

```diff
@@ -285,4 +289,52 @@ export class EducationProgramOfferingController {
       studyStartDate: offering.studyStartDate,
     };
   }
+
+  /**
+   * Get programs for a particular institution in paginated.
+   * @param institutionId id of the institution.
+   * @param take is the number of rows shown in the table
+   * @param skip is the number of rows that is skipped/offset from the total list.
+   * For example page 2 the skip would be 10 when we select 10 rows per page.
+   * @param dateSubmittedOrder the sorting order of the submitted date, default its ascending
+   * @param searchProgramName Search the program name in the query
+   * @returns ProgramsOfferingSummaryPaginated.
+   */
+  @AllowAuthorizedParty(AuthorizedParties.aest)
+  @Groups(UserGroups.AESTUser)
+  @Get("institution/:institutionId/programs/take/:take/skip/:skip")
+  async getPaginatedProgramsForInstitution(
+    @Param("institutionId") institutionId: number,
+    @Param("take") take: number,
+    @Param("skip") skip: number,
+    @Query("dateSubmittedOrder") dateSubmittedOrder: number,
+    @Query("searchProgramName") searchProgramName: string,
+  ): Promise<ProgramsOfferingSummaryPaginated> {
+    const paginatedProgramOfferingSummaryResult =
+      await this.programOfferingService.getPaginatedProgramsForInstitution(
+        institutionId,
+        take,
+        skip,
+        dateSubmittedOrder === -1 ? SortDBOrder.DESC : SortDBOrder.ASC,
```

Comment:
I would say that it would be clear, from API's perspective, if the parameters accepts ASC, DESC.
Is the -1 here defined because of the UI?

Reply:
Yes, from the UI the event returns -1 for DESC and 1 for ASC

Reply:
Please avoid adapting the API because of _whatever_ from the UI as much as possible.